hooks/useTopup.ts

```diff
@@ -1,23 +1,110 @@
-import { USDC_ADDRESS } from "@/lib/consts";
+import { USDC_ADDRESS, CHAIN_ID } from "@/lib/consts";
 import { useMemo, useState } from "react";
+import {
+  zeroAddress,
+  createWalletClient,
+  custom,
+  parseEther,
+  parseUnits,
+  erc20Abi,
+  Address,
+} from "viem";
+import { useSmartWalletProvider } from "@/providers/SmartWalletProvider";
+import useConnectedWallet from "./useConnectedWallet";
+import { useConnectWallet } from "@privy-io/react-auth";
+import getViemNetwork from "@/lib/viem/getViemNetwork";
+import { getPublicClient } from "@/lib/viem/publicClient";
+import { toast } from "sonner";
+
+export type DepositCurrency = "usdc" | "eth";
 
 const useTopup = () => {
-  const [depositAmount, setDepositAmount] = useState<string>("0.111");
+  const [depositAmount, setDepositAmount] = useState<string>("0.0111");
   const [isDepositing, setIsDepositing] = useState<boolean>(false);
+  const [currency, setCurrency] = useState<DepositCurrency>("usdc");
+
+  const { fetchSmartWallet, smartWallet } = useSmartWalletProvider();
+  const { externalWallet } = useConnectedWallet();
+  const { connectWallet } = useConnectWallet();
+
+  const currencyLabel = currency === "usdc" ? "USDC" : "ETH";
+  const hasExternalWallet = Boolean(externalWallet);
 
   const callData = useMemo(() => {
     return {
       totalPrice: depositAmount,
-      _usdc: USDC_ADDRESS,
+      _usdc: currency === "usdc" ? USDC_ADDRESS : zeroAddress,
     };
-  }, [depositAmount]);
+  }, [depositAmount, currency]);
+
+  const connectDisconnect = () => {
+    if (!hasExternalWallet) {
+      connectWallet();
+    } else {
+      externalWallet?.disconnect();
+    }
+  };
+
+  const deposit = async () => {
```

Comment:
Open Closed Principle
- actual: deposit function added inline to existing useTopup hook
- required: define the deposit function in a new, standalone, hook / lib and only if needed, import it into the useTopup hook.

<img width="676" height="240" alt="Image" src="https://github.com/user-attachments/assets/d9238f0a-00fb-41d3-a527-d65d61b9e6ab" />